### PR TITLE
feat(legacy): Added ChestStealer SmartOrder option.

### PR DESCRIPTION
**Pull request by doesDeveloper**
This feature optimizes chest stealing process to find the path which takes minimum time to loot the chest.

https://github.com/CCBlueX/LiquidBounce/assets/87639130/3191092d-3b94-4622-99e4-c1c21d60670e

### CODE REVIEW

**Comment by doesDeveloper** [comment]
Only issue this has is it has O(n^2) time complexity.
if someone with gud skill can reduce it to o(n) then it would be very helpful.

But its not that big of a deal cuz chests can have maximum of 54 itemstack. Not a big deal for today's cpu.

**Comment by doesDeveloper** [comment]
ready to merge if no issue ig.

**Comment by doesDeveloper** [comment]
when review?

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/ChestStealer.kt [hunk 1]
```diff
@@ -292,6 +292,26 @@ object ChestStealer : Module("ChestStealer", Category.WORLD, hideModule = false)
...
+                        next = itemsToSteal[j]
+                        nextIndex = j
+                    }
+                }
+                next?.let {
+                    itemsToSteal[nextIndex] = itemsToSteal[i+1]
+                    itemsToSteal[i+1] = next
+                }
+            }
+        }
```
Comment (superblaubeere27) [inline src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/ChestStealer.kt]: This function is already too long and too complex. This section should be it's own function

**Review by superblaubeere27 — COMMENTED** [review]
I don't think it is a really problem if stealing a *full* chest takes `n * (n - 1) / 2` steps if `n=57` (~1500 steps). The alternative is a quad tree or a k-d-tree both would take so long to construct that it isn't even worth it.

File: src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/ChestStealer.kt (hunk 1 above)
Comment (doesDeveloper) [reply src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/ChestStealer.kt]: ok

**Pull request merged**